translator/util/configutil.go

```diff
@@ -0,0 +1,67 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/aws/amazon-cloudwatch-agent/translator"
+	"github.com/aws/amazon-cloudwatch-agent/translator/config"
+)
+
+const (
+	source_ENV      = "env:"
+	source_FILE     = "file:"
+	sourceSeparator = ":"
+
+	yamlFileMode = 0644
+
+	linuxDownloadingPath   = "/opt/aws/amazon-cloudwatch-agent/etc/"
+	windowsDownloadingPath = "\\Amazon\\AmazonCloudWatchAgent\\"
+)
+
+func splitConfigPath(configPath string) (string, error) {
+	locationArray := strings.SplitN(configPath, sourceSeparator, 2)
+	if locationArray == nil || len(locationArray) < 2 {
+		return "", errors.New(fmt.Sprintf("config path: %s is malformatted.", configPath))
```

Comment:
We're technically changing behavior here since we're no longer panicking, but I prefer that to how it was before.

nit: Use `fmt.Errorf` instead of `errors.New(fmt.Sprintf)`
```suggestion
		return "", fmt.Errorf("config path: %s is malformatted.", configPath)
```

Reply:
panic is handled at config rule which is the caller